localstack/services/sns/sns_listener.py

```diff
@@ -637,6 +654,33 @@ def publish_message(topic_arn, req_data, headers, subscription_arn=None, skip_ch
     return message_id
 
 
+def publish_batch(topic_arn, req_data, headers):
+    response = {"Successful": [], "Failed": []}
+    messages = get_publish_batch_messages(req_data)
+    for message in messages:
+        message_id = str(uuid.uuid4())
+        data = {}
+        data["TopicArn"] = [topic_arn]
+        data["Message"] = [message["message_body"]]
+        data["Subject"] = [message["subject"]]
+        try:
+            start_thread(
```

Comment:
Let's not spawn a single thread per message, as this creates quite a large overhead and potentially negative performance impact. We could either (1) send all messages in a single thread, or (2) simply send the messages synchronously, without threading. In fact, if we want to return the list of `Successful` and `Failed` messages, we should use option (2), i.e., remove threading altogether from here.